hack/prepare.sh

```diff
@@ -25,6 +26,17 @@ function get_pf() {
 	readarray -t devs < <(devlink dev | awk -F/ '{print $2}')
 }
 
+function detect_card_type() {
+	local pf="${devs[0]}"
+	if devlink dev info pci/$pf | grep -q "fw.psid MT_0000000543"; then
```

Comment:
The changes look good! Just wondering if this PSID field needs to be set as configurable or at least to use a variable. As I can tell from this [website](https://docs.nvidia.com/networking/display/bluefield2firmwarev24401000/firmware+compatible+products), different serials of BlueField-2 cards (maybe with different prices), PSID could be slightly different.

Reply:
Valid point. I introduced an "array" of Bluefield identifiers now, so that "array" can be extended with the Bluefield identifiers that we want to support easily and the detection mechanism can stay the same.

Reply:
cool